flows/boundary.py

```diff
@@ -1296,8 +1296,11 @@ async def run_partial_updates_of_concepts_for_document_passages(
             f"Memory size (Mb): {text_blocks_size_in_mb}. "
         )
 
-        if grouped_concepts_n != text_blocks_n:
-            logger.warning(
+        # It's normal for there to be less text blocks from inference than vespa
+        # because inference filters out more types. However it should never be
+        # The other way around
+        if grouped_concepts_n > text_blocks_n:
+            raise ValueError(
```

Comment:
At this point in the project if the query returns 99,000 passages out of 100,000 do we not want to update concepts for the 99,000 we do have rather than fail here?

Reply:
It probably doesn't matter, unless we're able to re-run for just the 1,000. If we can't, then we'd have to run it on all of them anyway.